Merge recent changes into master and deploy to Pip

smholloway opened the pull request:
Time for a new version of the library on Pip! This pull request includes lots of good stuff, like 100% test coverage, a version constant, a unique user-agent, and zero-requester storage.

egrim committed:
Merge tag '1.0.5' into develop

t

* tag '1.0.5':
  Bump version number to 1.0.5

smholloway committed:
Add version constant to library, bump version, and test

smholloway committed:
Create a Toopher User-Agent string

smholloway committed:
Add headers to the mock request method to fix broken tests

smholloway committed:
Add Python version to User-Agent string

smholloway committed:
Update version number test to be more robust

smholloway committed:
Add Travis CI build status indicator

smholloway committed:
Merge pull request #15 from toopher/feature/add_version_number

Feature/add version number

trdarr committed:
Add QR pairing creation method

trdarr committed:
Add zero-storage

trdarr committed:
Remove response pprinting

trdarr committed:
Add tests

trdarr committed:
Update README

trdarr committed:
Comply with discussed nomenclature

trdarr committed:
Merge pull request #16 from toopher/feature/zero-storage

Add zero-storage

trdarr committed:
Feedback from @smholloway on PR 16

trdarr committed:
Update version number

Commit:
Fix some perl-specific readme verbiage

trdarr committed:
Replace python-oauth2 with Requests

trdarr committed:
Add requirements.txt

trdarr committed:
Pyflakes

trdarr committed:
Update tests to use Requests

trdarr committed:
Rename duplicate test

trdarr committed:
Add nose and coverage

trdarr committed:
Move api_url to module

trdarr committed:
Add AuthenticationStatusTests

trdarr committed:
Add PairingStatusTests

trdarr committed:
Remove unnecessary (?) status dicts

trdarr committed:
Add ZeroStorageTests

trdarr committed:
Move zero-storage tests

trdarr committed:
Improve test coverage (100%)

trdarr committed:
Use requirements.txt

trdarr committed:
Use variables instead of strings

trdarr committed:
disable ≠ enable

trdarr committed:
True ≠ False and vice versa

trdarr committed:
Add tests

trdarr committed:
Merge pull request #18 from toopher/fix-readme-verbiage

Fix some perl-specific readme verbiage

trdarr committed:
Sync with develop

* develop:

trdarr committed:
Sync with develop

* develop:
  Fix some perl-specific readme verbiage

trdarr committed:
Update setup.py dependencies

smholloway committed:
Merge pull request #17 from toopher/feature/zero-storage

Zero-Storage II: The Lost Years :+1:

trdarr commented:
> Time for a new version of the library on Pip!

https://speakerdeck.com/brettcannon/how-to-pronounce-pypi

egrim commented:
FYI: for language library releases I'll be in charge of actually merging into master (at least for now).  I would still like go/no-go decisions from reviewers though.

smholloway commented:
:+1:  I'm a go.

trdarr commented:
:goat: I’m a go(at).

smholloway commented:
We've found a bug with certificate validation. Please hold off on merging until we resolve the issue.

trdarr committed:
Verify SSL, actually

trdarr commented:
We’ve fixed the bug with certificate validation: #20.

trdarr committed:
Merge pull request #20 from toopher/bugfix/verify-ssl

Fix SSLError; actually verify SSL

smholloway committed:
README updates for zero-storage

Uppercased true and false; updated the zero-storage exception names; added and fixed links; updated the Dependencies section based on recent changes

smholloway commented:
TR's pull request fixes the issue I mentioned a couple hours ago. I'm back to a goat :goat: .

smholloway committed:
Convert to a lowercase string comparison

While testing the PR I found the error message was ToopherApiError: Not allowed: Pairing has not been authorized to authenticate, so Pairing did not match pairing. This should fix the case sensitive search issue.

smholloway committed:
Update the error message to match the server response

The server responds with capitalized Pairing, which necessitates a case insensitive search (already committed).

trdarr committed:
Use ye olde '%' formatting

trdarr committed:
Test against Python 2.5 (instead of 2.7)

trdarr committed:
Test against Python 2.6 instead of 2.5

trdarr committed:
Use 2.6-compatible unittest.assertRaises calls

trdarr committed:
Use assertTrue instead of assertGreaterEqual

trdarr committed:
Merge pull request #22 from toopher/feature/percent-format

Support Python 2.6

egrim committed:
Fix __getattr__ implementation to raise correct exception

If an unknown attribute name access was attempted on `PairingStatus` or `AuthenticationStatus` objects they would raise a KeyError instead of an AttributeError.  This breach of contract breaks things like pickling - so this commit fixes that.

egrim committed:
Merge pull request #23 from toopher/bug/getattr-breach-of-contract

Fix __getattr__ implementation to raise correct exception

egrim committed:
Fix the tests that verified the breach-of-contract for __getattr__

egrim committed:
Merge pull request #24 from toopher/bug/getattr-breach-of-contract

Fix the tests that verified the breach-of-contract for __getattr__

egrim committed:
Fix infinite recursion when unpickling PairingStatus and AuthenticationStatus

egrim committed:
Merge pull request #25 from toopher/bug/getattr-breach-of-contract

Fix infinite recursion when unpickling PairingStatus and AuthenticationStatus

Commit:
Add iframe libs

Commit:
PR feedback - make return type consistent with method name

Commit:
PR feedback

Commit:
PR Feedback - this is why we have code review, people

Commit:
update to reflect new pairing deactivated error code 707

Commit:
Add iframe README

Commit:
Deal with params passed in as dict-of-lists

trdarr committed:
Add tests for exceptions

dshafer committed:
Merge pull request #27 from toopher/feature/iframe

Add iframe libs

trdarr committed:
Require coveralls

trdarr committed:
Travis? Travis!

trdarr committed:
Ignore non-Toopher files

smholloway committed:
Change Java comment to Python comment

smholloway committed:
Convert to fenced code blocks w language specified

smholloway committed:
Fix namespace of errors

They're in the `toopher` module--not the `ToopherIframe`

smholloway committed:
Convert from Java error to Python error

toopher-python: ERROR_CODE_USER_DISABLED = 704
toopher-java: public static final String USER_OPT_OUT = "704";

smholloway committed:
Test validate with immutable dict

smholloway committed:
Create a mutable copy of data to satisfy test

smholloway committed:
Remove debug breakpoint

smholloway committed:
Ignore the tests file

smholloway committed:
Switch to using the new web/authenticate endpoint

smholloway committed:
Merge pull request #30 from toopher/issue/29-make-data-mutable

Issue/29 make data mutable

smholloway committed:
Merge develop

smholloway committed:
Merge pull request #26 from toopher/feature/qr

Add pair_qr for generating Toopher QR codes

smholloway committed:
Merge develop

smholloway committed:
Merge pull request #28 from toopher/feature/coveralls

Add Coveralls support

smholloway committed:
Merge pull request #31 from toopher/feature/default-to-all-in-one-iframe-endpoint

Switch to using the new `web/authenticate` endpoint

Commit:
add helper method for manage_user iframe endpoint

smholloway committed:
Remove spaces in default parameter

dshafer committed:
Merge pull request #34 from toopher/feature/manage_user_iframe_endpoint

add helper method for manage_user iframe endpoint

Commit:
modify tests for allow_inline_pairing parameter in iframe

Commit:
add allow_inline_pairing kwarg to authenticate iframe

dshafer committed:
Merge pull request #35 from toopher/feature/manage_user_iframe_endpoint

Add kwarg to specify allow_inline_pairing in auth iframe

Commit:
Add CONTRIBUTING.md

Commit:
Add CONTRIBUTING.md

Commit:
Sync changes with branch 'add-contributing-readme'

Commit:
Add Python version

smholloway committed:
Merge pull request #36 from toopher/add-contributing-readme

Add CONTRIBUTING.md
